desktop/docs/electron_architecture.md

```diff
@@ -0,0 +1,114 @@
+## Electron Architecture
+
+### Pinentry
+
+Pinentry will be handled separately from the other desktop MVP features, through
+a new `electron-password` application, which:
+
+1. reads JSON instructions on stdin until EOF
+1. spawns a single window
+1. inside that window, renders HTML from a given path (which can include JS)
+1. prints the user input it received as JSON to stdout
+1. exits
+
+Example stdin:
+```js
+{
+  "html_path": "/usr/local/share/keybase/desktop/templates/password.html",
+  "window_title": "Keybase",
+  "prompt_text": "Please enter your passphrase to unlock your private key",
+  "checkbox": true,
+  "checkbox_text": "Store my passphrase",
+}
```

Comment:
how about multiple checkboxes, or maybe a radio box eventually for "save my passphrase indefinitely", "save it for 5 hours", or "not at all"?

we don't have to build this at first but the JSON input/output should be able to accommodate it.